lib/openhab/dsl/rules/automation_rule.rb

```diff
@@ -253,12 +262,18 @@ def check_guards(event:)
         #
         #
         def process_queue(run_queue, mod, inputs)
-          event = inputs&.dig('event')
-
           while (task = run_queue.shift)
             if task.is_a? RuleConfig::Delay
               process_delay_task(inputs, mod, run_queue, task)
             else
+              event = inputs&.dig('event')
+              attachment = attachment(inputs)
+              if attachment
+                class << event
+                  attr_reader :attachment
+                end
+                event.instance_variable_set('@attachment', attachment)
+              end
```

Comment:
Is there a reason why this block of code needs to be inside the `while` loop? It seems to me that it could be done once before the loop.